[Warnings] Make [p]warnings usable on base of permissions

Dav-Git opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Makes [p]warnings work with permissions and introduces [p]mywarnings with the functionality described in #2900

Dav-Git committed:
new code

Added the admin check to warnings and removed the part where the user could check themselves.

Added new mywarns which replaces part of the old behaviour of warn

Dav-Git committed:
Update warnings.py

Dav-Git committed:
Create 2900.enhance.rst

Dav-Git committed:
Merge branch 'fix-warnings-for-mods' of https://github.com/Dav-Git/Dav-patches into fix-warnings-for-mods

Jackenmen commented:
Changelog contains wrong command names + I think changelogs should be separated into 2 files - 2900.enhance.1.rst and 2900.enhance.2.rst (though I heard today it might not be necessary so someone please confirm this)

Dav-Git committed:
Fixed command name

Because appearently I can't remember a command for 10 seconds

Dav-Git commented:
Wasn't sure on 2 files. I thought 1 file works

Dav-Git committed:
Commands in backticks

Put command names in changelog in double backticks after being advised to do so in discord

mikeshardmind commented on redbot/cogs/warnings/warnings.py:
```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+            await ctx.send(_("Please provide a user you want to check the warnings for."))
```
This should either return here, or be removed entirely and remove the optional nature of the user argument (probably the latter, but either would be fine)

Dav-Git replied:
Would removing the "Optional" part of the argument fix this?

jack1142 replied:
Yes, making the argument required and removing handling for `user is None` would fix this.

Jackenmen replied:
Yes, making the argument required and removing handling for `user is None` would fix this.
(adding again, cause I accidentally removed the comment)

mikeshardmind commented on redbot/cogs/warnings/warnings.py:
```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
+            await ctx.send(_("Please provide a user you want to check the warnings for."))
+
+        msg = ""
+        member_settings = self.config.member(user)
```
This doesn't properly account for the possibility a user id was passed without it being to a current user (and needs to)

Dav-Git replied:
Honestly I have no idea how to go about this.

Jackenmen replied:
The removed handling here was doing that before:
https://github.com/Cog-Creators/Red-DiscordBot/pull/3327/files#diff-bd66023271557007376afcf648f3b3ffL363-L368

Dav-Git committed:
made user not optional, and the other thing sinbad requested

Dav-Git commented:
I think that's what you had in mind?

mikeshardmind commented on redbot/cogs/warnings/warnings.py:
```diff
@@ -342,36 +341,52 @@ def __init__(self, bot: Red):
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+
             try:
```
Not exactly what I meant... 

The git diff here is doing some slight disfavors in discussing review as it's recognizing the code for `mywarnings` now matches the old `warnings` more closely.

The handling for if a user id not a user was passed isn't needed in `mywarnings` but is needed in `warnings` (prior to interacting with config)

Dav-Git committed:
switched parts. magic

mikeshardmind reviewed (approved):
Thanks!
